src/main/scala/org/allenai/plugins/VersionInjectorPlugin.scala

```diff
@@ -117,6 +140,51 @@ object VersionInjectorPlugin extends AutoPlugin {
         gitConfFile
     }
 
+  val injectCacheKeyTask = injectCacheKey := {
+    val values = ((resourceManaged in Compile).value.absolutePath, organization.value, name.value, streams.value, cacheKey.value)
+    val (resourceManagedPath, org, nameValue, s, cacheKeyResult) = values
+    val cacheKeyConfFile = new java.io.File(s"$resourceManagedPath/$org/${cleanArtifactName(nameValue)}/cacheKey.conf")
+    s.log.info(s"Generating cacheKey.conf managed resource... (cacheKey: $cacheKeyResult)")
+
+    val cacheKeyContents = s"""cacheKey: "$cacheKeyResult""""
+    IO.write(cacheKeyConfFile, cacheKeyContents)
+    cacheKeyConfFile
+  }
+
+  //Gives us the git most recent commits for all the local projects that this project depends on
+  lazy val gitMRCs = Def.taskDyn { // We have to use a dynamic task because generating tasks based on project dependencies
+    // necessarily alters the task graph
+    val MRCs = buildDependencies.value.classpathRefs(thisProjectRef.value).sorted // get the local dependencies
+    val filter = ScopeFilter(inProjects(MRCs: _*)) // this is weird, we create a scopefilter on the dependencies
+    gitLocalSha1.all(filter) // odd piece of syntax - returns a list of tasks (we're inside a taskDyn block) that is
+    //applying gitLocalSHa1 to all scopes in the scopefilter 'filter'
+  }
+
+  val cacheKeyTask = cacheKey := {
+    import java.io.{ File => JFile }
+
+    def getFileList(dir: String): Seq[JFile] = new JFile(dir) match {
+      case f if f.exists && f.isDirectory => f.listFiles.filter(_.isFile).toSeq
+      case _ => Seq.empty
+    }
+    val stageDir = baseDirectory.value + "/target/universal/stage"
+    val allFiles: Seq[JFile] = Seq("bin", "conf", "lib", "public")
+      .map { dir => stageDir + "/" + dir }
+      .flatMap(getFileList)
+
+    val filesToHash = allFiles filterNot { f: JFile =>
+      //TODO: make this a setting
+      val fileName = f.getName
+      fileName.startsWith("org.allenai.ari-api") ||
+        fileName.startsWith("org.allenai.solvers-") ||
+        fileName.startsWith("org.allenai.ari-solvers-")
+    }
+    val hashes = filesToHash.sorted
+      .map(Hash.apply)
+      .map(Hash.toHex)
+    Hash.toHex(Hash((hashes ++ gitMRCs.value :+ gitLocalSha1.value).sorted.mkString))
```

Comment:
This is the only place you need to have a `sorted` call. You should add a comment, too, so that the sorting doesn't get deleted by an over-zealous refactorer later on . . .